packages/ui/src/components/Avatar/Avatar.tsx

```diff
@@ -14,48 +14,72 @@
  * limitations under the License.
  */
 
-import { forwardRef, ElementRef } from 'react';
-import { Avatar as AvatarPrimitive } from '@base-ui-components/react/avatar';
+import { forwardRef, useState, useEffect } from 'react';
 import clsx from 'clsx';
 import { AvatarProps } from './types';
 import { useStyles } from '../../hooks/useStyles';
 import styles from './Avatar.module.css';
 
 /** @public */
-export const Avatar = forwardRef<
-  ElementRef<typeof AvatarPrimitive.Root>,
-  AvatarProps
->((props, ref) => {
+export const Avatar = forwardRef<HTMLDivElement, AvatarProps>((props, ref) => {
   const { classNames, dataAttributes, cleanedProps } = useStyles('Avatar', {
     size: 'medium',
+    purpose: 'informative',
     ...props,
   });
 
-  const { className, src, name, ...rest } = cleanedProps;
+  const { className, src, name, purpose, ...rest } = cleanedProps;
+
+  const [imageStatus, setImageStatus] = useState<
+    'loading' | 'loaded' | 'error'
+  >('loading');
+
+  useEffect(() => {
+    setImageStatus('loading');
+    const img = new Image();
+    img.onload = () => setImageStatus('loaded');
+    img.onerror = () => setImageStatus('error');
+    img.src = src;
+
+    return () => {
+      img.onload = null;
+      img.onerror = null;
+    };
+  }, [src]);
+
+  const initials = name
+    .split(' ')
+    .map(word => word[0])
+    .join('')
+    .toLocaleUpperCase('en-US')
+    .slice(0, 2);
 
   return (
-    <AvatarPrimitive.Root
+    <div
       ref={ref}
+      role="img"
+      aria-label={purpose === 'informative' ? name : undefined}
+      aria-hidden={purpose === 'decoration' ? true : undefined}
```

Comment:
This feels a little bit clunky somehow, feels enough for callers to just set `aria-hidden`, or do we need to make sure the label is unset for that case as well? Or `role=presentation`? although not sure about that route